packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useCreateFolderPath.ts

```diff
@@ -0,0 +1,68 @@
+import { useMutation } from '@tanstack/react-query'
+import { allowNotFoundError } from '../../../synapse-client/SynapseClientUtils'
+import { useCreateEntity } from '../../../synapse-queries/index'
+import { useSynapseContext } from '../../context/index'
+import {
+  convertToEntityType,
+  entityTypeToFriendlyName,
+} from '../../functions/EntityTypeUtils'
+
+/**
+ * Given the ID of a container entity and a path string array, create the folder structure as necessary and return the ID of the final folder in the path.
+ * @returns The ID of the final folder in the path.
+ */
+export function useCreateFolderPath() {
```

Comment:
When uploading an entire folder, we may need to create many nested folders. This hook handles creating those folders.